packages/core/src/internal/sagaHelpers/debounce.js

```diff
@@ -1,8 +1,16 @@
 import fsmIterator, { safeName } from './fsmIterator'
 import { delay, fork, race, take } from '../io'
 
-export default function debounceHelper(delayLength, patternOrChannel, worker, ...args) {
-  let action, raceOutput
+export default function debounceHelper(delayLengthOrOptions, patternOrChannel, worker, ...args) {
```

Comment:
What happens if both `leading` and `trailing` are set to `true`?

Reply:
The saga will run at the start of the wait period, and if any further actions occur during the wait period, it will run again at the end of the wait period. This matches the behavior of the lodash debounce function with `leading: true, trailing: true`. I imagine this will be a common use-case - it's how I will use it personally.

There is an example of this in the tests.

Reply:
I have deployed a little example here https://heartfelt-semifreddo-a4737f.netlify.app/
(Hit any key on the keyboard to see the effect of a 1000ms debounce with the various options).

Source code is [here](https://github.com/ensconced/redux-saga-trailing-leading-demo/blob/master/index.js)